docs/src/index.md

```diff
@@ -29,7 +29,52 @@ F(t,y) = y
 The vectors `t` and `y` store the time and solution values at the
 corresponding times.
 
-You might find the basic interface limiting.  First of all, it stores
+## Solve interface
+
+`ODE.ode` only supports explicit differential equations defined as
+`y'=F(t,y)`, for more advenced uses consider using `ODE.solve`, which
+was designed to work with a variety of other types of initial value
+problems and is optimized for better performance.  First we have to
+define an initial value problem, in our case this is an explicit
+differential equation `y'=y` with inital data `y0=[1.0]` given at the
+time `t0=0.0`.
+
+```@example solve
+using ODE
+t0  = 0.0
+y0  = [1.0]
+F!(t,y,dy) = dy[1]=y[1]
+ode = ODE.ExplicitODE(t0,y0,F!)
+```
+
+Note that unlike in `ODE.ode` we now have to supply an in place
+function `F!` instead of an explicit function `F`.  We can solve the
+ODE problem `ode` by simply calling
+
+```@example solve
+sol = ODE.solve(ode, tstop = 1)
+```
```

Comment:
It looks weird to me to have a function mixing high level arguments like `IVP` with simple numbers in a single call: `solve(ivp, 1)`.  It is way clearer what the number means if we write it as `solve(ivp,tstop=1)` and error on `!isfinite`.  It also leaves us more flexibility in how to treat the infinite `tstop` case.  And as I already mentioned, strictly speaking `tstop` is not the only way to limit the integration.